File: app/src/main/java/com/song/deviceinfo/ui/system/SystemFragment.java
package com.song.deviceinfo.ui.system;

import com.song.deviceinfo.ui.base.BaseAdapter;
import com.song.deviceinfo.ui.base.BaseViewModel;
import com.song.deviceinfo.ui.base.NormalAdapter;
import com.song.deviceinfo.ui.base.NormalFragment;

import java.util.List;

import androidx.core.util.Pair;
import androidx.lifecycle.ViewModelProviders;

/**
 * Created by chensongsong on 2020/5/27.
 */
public class SystemFragment extends NormalFragment {

    @Override
    protected List<Pair<String, String>> getNormalInfo() {
        return ((SystemViewModel) viewModel).getSystemInfo(getActivity());
    }

    @Override
    protected BaseAdapter crateAdapter() {
        return new NormalAdapter(getContext());
    }

    @Override
    protected BaseViewModel crateViewModel() {
        return ViewModelProviders.of(this).get(SystemViewModel.class);
    }
}

File: app/src/main/java/com/song/deviceinfo/ui/system/SystemFragment.kt
package com.song.deviceinfo.ui.system

import androidx.core.util.Pair
import androidx.lifecycle.ViewModelProviders
import com.song.deviceinfo.ui.base.BaseAdapter
import com.song.deviceinfo.ui.base.BaseViewModel
import com.song.deviceinfo.ui.base.NormalAdapter
import com.song.deviceinfo.ui.base.NormalFragment

/**
 * Created by chensongsong on 2020/5/27.
 */
class SystemFragment : NormalFragment() {
    override val normalInfo: List<Pair<String, String>>
        get() = (viewModel as SystemViewModel).getSystemInfo(requireActivity())
    
    override fun createAdapter(): BaseAdapter<Pair<String, String>, *> {
        return NormalAdapter(requireContext())
    }
    
    override fun createViewModel(): BaseViewModel<Pair<String, String>> {
        return ViewModelProviders.of(this).get(SystemViewModel::class.java)
    }
}
